src/common/filter/Expressions.cpp

```diff
@@ -839,10 +842,22 @@ OptVariantType ArithmeticExpression::eval() const {
             }
             break;
         case MOD:
-            if (isInt(l) && isInt(r)) {
+            if (isArithmetic(l) && isArithmetic(r)) {
+                if (isDouble(l) || isDouble(r)) {
+                    return fmod(asDouble(l), asDouble(r));
+                }
                 return OptVariantType(asInt(l) % asInt(r));
             }
             break;
+        case XOR:
+            if (isArithmetic(l) && isArithmetic(r)) {
+                if (isDouble(l) || isDouble(r)) {
```

Comment:
Do we really need to support xor for double? It is a little bit weird that we round up double and then do xor.

Reply:
I see mysql is supported

Reply:
What scenarios are applicable for xor with double ?

Reply:
@critical27 @darionyaphet 
Personally, I have not seen this usage.

But from the perspective of semantic of XOR itself, or other bit ops like bitwise AND, they are just literally performing the ops bitwisely. It's up to the user to the determine whether the op is meaningful.